File: app/src/main/java/com/example/sergejmitrofanov/librarycatalogue/interactor/BookListTitleSorter.java
package com.example.sergejmitrofanov.librarycatalogue.interactor;

import androidx.annotation.NonNull;
import com.example.sergejmitrofanov.librarycatalogue.entity.Book;
import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

public class BookListTitleSorter implements BookListSorter {

  @SuppressWarnings("ComparatorCombinators")
  @NonNull
  @Override
  public List<Book> sortBooks(@NonNull List<Book> books) {
    List<Book> sortedBooks = new ArrayList<>(books);
    Collections.sort(sortedBooks, (o1, o2) -> o1.bookTitle.compareTo(o2.bookTitle));
    return sortedBooks;
  }
}

File: app/src/main/java/com/example/sergejmitrofanov/librarycatalogue/interactor/BookListTitleSorter.kt
package com.example.sergejmitrofanov.librarycatalogue.interactor

import com.example.sergejmitrofanov.librarycatalogue.entity.Book
import javax.inject.Inject

class BookListTitleSorter @Inject constructor() : BookListSorter {
    override fun sortBooks(books: List<Book>) = books.sortedBy { it.bookTitle }
}
